### PR TITLE
Simplify Enroll User API

### PR DESCRIPTION
**Pull request by mbyczkowski**
PR #82 introduced an endpoint for issuing SSH user certificates. The API
assumed that the user param passed in the API should match the username
in the HTTP header passed by the authenticating proxy. Since, there's no
functionality, currently, that allows a user to issue a cert on behalf
of another user (e.g. authenticated user, Alice, trying to call
/enroll_user/bob), this part of the API is redundant and is ripe for
simplification.

This PR is just #84 leftovers 😉

### CODE REVIEW

File: server/enroll.go [hunk 1]
```diff
@@ -157,37 +157,29 @@ func (c *context) sign(keyId string, principals []string, serial uint64, certTyp
 // This assumes there's an authenticating proxy which provides the user in a header, configurable.
 // We identify the proxy with its TLS client cert
 // TODO: this function needs test coverage
-func proxyAuthenticated(ap *config.AuthenticatingProxy, requestedUser string, w http.ResponseWriter, r *http.Request) bool {
+func proxyAuthenticated(ap *config.AuthenticatingProxy, userAuthed string, w http.ResponseWriter, r *http.Request) bool {
```
Comment (mcpherrinm) [inline server/enroll.go]: I think this function should be changed to return the user, rather than reading the header in the caller & passing it in

Comment (mcpherrinm) [reply server/enroll.go]: That ensures all the proxy handling code (mtls requirement + header handling) are in one single atomic testable unit

Comment (mbyczkowski) [reply server/enroll.go]: I've rewritten this a bit and added more tests. How's that now?

Comment (mcpherrinm) [reply server/enroll.go]: looks great now, thank you!

**Pull request merged**